pool/write.go

```diff
@@ -0,0 +1,100 @@
+package pool
+
+import (
+	"bytes"
+	"time"
+
+	"github.com/lightningnetwork/lnd/buffer"
+)
+
+// Write is a worker pool specifically designed for sharing access to
+// buffer.Write objects amongst a set of worker goroutines. This enables an
+// application to limit the total number of buffer.Write objects allocated at
+// any given time.
+type Write struct {
+	workerPool *Worker
+	bufferPool *WriteBuffer
+}
+
+// NewWrite creates a Write pool, using an underlying Writebuffer pool to
+// recycle buffer.Write objects accross the lifetime of the Write pool's
+// workers.
+func NewWrite(writeBufferPool *WriteBuffer, numWorkers int,
+	workerTimeout time.Duration) *Write {
+
+	w := &Write{
+		bufferPool: writeBufferPool,
+	}
+	w.workerPool = NewWorker(&WorkerConfig{
+		NewWorkerState: w.newWorkerState,
+		NumWorkers:     numWorkers,
+		WorkerTimeout:  workerTimeout,
+	})
+
+	return w
+}
+
+// Start safely spins up the Write pool.
+func (w *Write) Start() error {
+	return w.workerPool.Start()
+}
+
+// Stop safely shuts down the Write pool.
+func (w *Write) Stop() error {
+	return w.workerPool.Stop()
+}
+
+// Submit accepts a function closure that provides access to a fresh
+// bytes.Buffer backed by a buffer.Write object. The function's execution will
+// be allocated to one of the underlying Worker pool's goroutines.
+func (w *Write) Submit(inner func(*bytes.Buffer) error) error {
+	return w.workerPool.Submit(func(s WorkerState) error {
+		state := s.(*writeWorkerState)
```

Comment:
Just to be on the safe side, I think we should check the second value and error our here to avoid a panic if it isn't our expected type. This is unlikely to happen, but if we check the value, then a bug in a lower level of abstraction won't cause lnd to crash all together.

Reply:
i figured that was the purpose of introducing the unit tests?